s3/src/main/java/ch/cyberduck/core/s3/S3AttributesAdapter.java

```diff
@@ -83,7 +83,7 @@ public String getDescription() {
             attributes.setMetadata(metadata);
             if(object.containsMetadata(S3TimestampFeature.METADATA_MODIFICATION_DATE)) {
                 try {
-                    attributes.setModificationDate(Double.valueOf(object.getMetadata(S3TimestampFeature.METADATA_MODIFICATION_DATE).toString()).longValue());
+                    attributes.setModificationDate(Double.valueOf(object.getMetadata(S3TimestampFeature.METADATA_MODIFICATION_DATE).toString()).longValue() * 1000);
```

Comment:
Does this affect how Cyberduck persists Mtime as well?
That is timestamps uploaded through Cyberduck are now off?

Reply:
We are setting the value in millis 🫣 [^1] 

[^1]: https://github.com/iterate-ch/cyberduck/blob/e990d4c43c32e6246bc403b617b45bf0742c524e/s3/src/main/java/ch/cyberduck/core/s3/S3WriteFeature.java#L140-L143